File: app/src/main/java/com/papco/sundar/papcortgs/database/receiver/Receiver.java
package com.papco.sundar.papcortgs.database.receiver;

import android.text.SpannableString;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity
public class Receiver {

    @PrimaryKey(autoGenerate = true)
    public int id;
    public String accountType;
    public String accountNumber;
    public String name;
    public String displayName;
    public String mobileNumber;
    public String ifsc;
    public String bank;
    public String email="";
    @Ignore
    public SpannableString highlightedName;
}

File: app/src/main/java/com/papco/sundar/papcortgs/database/receiver/Receiver.kt
package com.papco.sundar.papcortgs.database.receiver

import androidx.core.text.isDigitsOnly
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class Receiver {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    var accountType: String = ""
    var accountNumber: String = ""
    var name: String = ""
    var displayName: String = name
    var mobileNumber: String = ""
    var ifsc: String = ""
    var bank: String = ""
    var email = ""

    fun hasValidMobileNumber(): Boolean {
        if (mobileNumber.trim { it <= ' ' }.length != 10) return false
        return mobileNumber.isDigitsOnly()
    }
}

